core/plan/dockerignore.go

```diff
@@ -0,0 +1,88 @@
+package plan
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/moby/patternmatcher/ignorefile"
+)
+
+// checks if a .dockerignore file exists in the app directory and parses it
+func CheckAndParseDockerignore(repoPath string) ([]string, []string, error) {
```

Comment:
Thoughts on passing in the app directly here? All filesystem reads up until now (should) go through the app abstraction and keeping that in place has future benefits IMO (we could read from a GitHub repo using the API or directly from a compressed directory). Far out, but if we are only reading the file once in this function, it is simple enough to keep that abstraction in place.

Reply:
good call, just refactored this